Gems/Atom/RPI/Code/Include/Atom/RPI.Public/GpuQuery/GpuPassProfiler.h

```diff
@@ -0,0 +1,132 @@
+        class ParentPass;
+
+        //! This is a helper class that can be used to measure, per frame, the time it takes
+        //! for the render pipeline to execute in the GPU from the first to the last pass.
+        //! The core functionality is provided by the function MeasureRenderPipelineGpuTimeInNanoseconds(),
+        //! but other functions are available for tools who need to report more details, like time spent at each pass, etc.
+        //! REMARK: Use judiciously, as MeasureRenderPipelineGpuTimeInNanoseconds() can itself affect performance when called.
+        //! For example, if a scene is running at 300fps, calling MeasureRenderPipelineGpuTimeInNanoseconds(), each frame, can reduce the FPS to
+        //! ~265fps.
+        class GpuPassProfiler final
+        {
+        public:
+
+            //! Intermediate data that represents the structure of a Pass within the FrameGraph.
+            //! A tree structure will be created from these entries that mimics the pass' structure.
+            //! By default, all entries have a parent<-child reference, but only entries that pass the filter will also hold a parent->child reference.
+            struct PassEntry
+            {
+                //! Total number of attributes collected per  PipelineStatistics.
+                static const uint32_t PipelineStatisticsAttributeCount = 7u;
+
+                using PipelineStatisticsArray = AZStd::array<uint64_t, PipelineStatisticsAttributeCount>;
+
+                PassEntry() = default;
+                PassEntry(const class AZ::RPI::Pass* pass, PassEntry* parent);
+
+                ~PassEntry() = default;
+
+                //! Links the child TimestampEntry to the parent's, and sets the dirty flag for both the parent and child entry.
+                //! Calling this method will effectively add a parent->child reference for this instance, and all parent entries leading up to this
+                //! entry from the root entry.
+                void LinkChild(PassEntry* childEntry);
+
+                //! Checks if timestamp queries are enabled for this PassEntry.
+                bool IsTimestampEnabled() const;
+                //! Checks if PipelineStatistics queries are enabled for this PassEntry.
+                bool IsPipelineStatisticsEnabled() const;
+
+                //! The name of the pass.
+                Name m_name;
+                //! Cache the path name of the Pass as a unique identifier.
+                Name m_path;
+
+                RPI::TimestampResult m_timestampResult;
+                uint64_t m_interpolatedTimestampInNanoseconds = 0u;
+
+                //! Convert the PipelineStatistics result to an array for easier access.
+                PipelineStatisticsArray m_pipelineStatistics;
+
+                //! Used as a double linked structure to reference the parent <-> child relationship.
+                PassEntry* m_parent = nullptr;
+                AZStd::vector<PassEntry*> m_children;
```

Comment:
The pass tree structure is not used by the GpuPassProfiler's functions (only created but not used). Do you keep it in case to switch the ImGuiProfiler's implementation to use this class?